tests/test_cli.py

```diff
@@ -883,3 +919,64 @@ def test_check_pod_is_missing_session(self, session, sample_serial_workflow_in_d
             assert len(in_sync) == 0
             assert len(out_of_sync) == 0
             assert len(pods_without_session) == 1
+
+    def test_check_workspaces(
+        self, sample_serial_workflow_in_db, tmp_path: pathlib.Path
+    ):
+        workflow_id = str(sample_serial_workflow_in_db.id_)
+        user_id = str(sample_serial_workflow_in_db.owner.id_)
+
+        # prepare wrong workspace
+        extra_workspace_path = tmp_path.joinpath(
+            "users", user_id, "workflows", workflow_id
+        )
+        extra_workspace_path.mkdir(parents=True)
+
+        with patch(
+            "reana_server.reana_admin.check_workflows.SHARED_VOLUME_PATH",
+            str(tmp_path),
+        ):
+            extra_workspaces = check_workspaces()
+
+        assert len(extra_workspaces) == 1
+
+        result = extra_workspaces[0]
+        assert result.source.workspace == str(extra_workspace_path)
+        assert result.source.id == workflow_id
+        assert result.source.user == sample_serial_workflow_in_db.owner.email
+        assert result.source.name == sample_serial_workflow_in_db.name
+        assert len(result.errors) == 2
+        assert any("not owned" in str(error) for error in result.errors)
+        assert any(
+            sample_serial_workflow_in_db.workspace_path in str(error)
+            for error in result.errors
+        )
+
+    @pytest.mark.parametrize(
+        "user_id, workflow_id",
+        [(generate_uuid(), generate_uuid()), ("random-user", "random-workflow")],
+    )
+    def test_check_workspaces_without_related_owner(
```

Comment:
Genuinely asking, why did you separate this test from the previous one? Is it to keep the case in which there is an owner in the DB  logically distinct from the one in which there is not?

Reply:
To be honest, because I have written them not at the same time. I will join them in the same test